tests/unittests/tests-nanocoap_cache/tests-nanocoap_cache.c

```diff
@@ -0,0 +1,275 @@
+    char *path = "/path";
+    nanocoap_cache_entry_t *c = NULL;
+    size_t len;
+    int res;
+
+    /* initialize the nanocoap cache */
+    nanocoap_cache_init();
+
+    TEST_ASSERT_EQUAL_INT(CONFIG_NANOCOAP_CACHE_ENTRIES,
+                          nanocoap_cache_free_count());
+    TEST_ASSERT_EQUAL_INT(0, nanocoap_cache_used_count());
+
+    /* request */
+    len = coap_build_hdr((coap_hdr_t *)&buf[0], COAP_TYPE_NON,
+                         &token[0], 2, COAP_METHOD_GET, msgid);
+    coap_pkt_init(&req, &buf[0], sizeof(buf), len);
+    coap_opt_add_string(&req, COAP_OPT_URI_PATH, &path[0], '/');
+    coap_opt_finish(&req, COAP_OPT_FINISH_NONE);
+
+    /* response */
+    len = coap_build_hdr((coap_hdr_t *)&rbuf[0], COAP_TYPE_NON,
+                         &token[0], 2, COAP_CODE_205, msgid);
+    coap_pkt_init(&resp, &rbuf[0], sizeof(rbuf), len);
+    coap_opt_finish(&resp, COAP_OPT_FINISH_NONE);
+
+    c = nanocoap_cache_add_by_req((const coap_pkt_t *)&req,
+                                  (const coap_pkt_t *)&resp,
+                                  CONFIG_NANOCOAP_CACHE_RESPONSE_SIZE);
+    TEST_ASSERT_NOT_NULL(c);
+
+    TEST_ASSERT_EQUAL_INT(CONFIG_NANOCOAP_CACHE_ENTRIES - 1,
+                          nanocoap_cache_free_count());
+    TEST_ASSERT_EQUAL_INT(1, nanocoap_cache_used_count());
+
+    /* delete previously added cache entry */
+    res = nanocoap_cache_del(c);
+    TEST_ASSERT_EQUAL_INT(0, res);
+    TEST_ASSERT_EQUAL_INT(CONFIG_NANOCOAP_CACHE_ENTRIES,
+                          nanocoap_cache_free_count());
+    TEST_ASSERT_EQUAL_INT(0, nanocoap_cache_used_count());
+}
+
+static void test_nanocoap_cache__max_age(void)
+{
+    uint8_t buf[_BUF_SIZE];
+    uint8_t rbuf[_BUF_SIZE];
+    coap_pkt_t req, resp;
+
+    uint16_t msgid = 0xABCD;
+    uint8_t token[2] = {0xDA, 0xEC};
+    char *path = "/path";
+    nanocoap_cache_entry_t *c = NULL;
+    size_t len;
+    uint64_t now;
+
+    /* initialize the nanocoap cache */
+    nanocoap_cache_init();
+
+    /* request */
+    len = coap_build_hdr((coap_hdr_t *)&buf[0], COAP_TYPE_NON,
+                         &token[0], 2, COAP_METHOD_GET, msgid);
+    coap_pkt_init(&req, &buf[0], sizeof(buf), len);
+    coap_opt_add_string(&req, COAP_OPT_URI_PATH, &path[0], '/');
+    coap_opt_finish(&req, COAP_OPT_FINISH_NONE);
+
+    /* response with max-age 30 sec */
+    len = coap_build_hdr((coap_hdr_t *)&rbuf[0], COAP_TYPE_NON,
+                         &token[0], 2, COAP_CODE_205, msgid);
+    coap_pkt_init(&resp, &rbuf[0], sizeof(rbuf), len);
+    coap_opt_add_uint(&resp, COAP_OPT_MAX_AGE, 30);
+    coap_opt_finish(&resp, COAP_OPT_FINISH_NONE);
+
+    c = nanocoap_cache_add_by_req((const coap_pkt_t *)&req,
+                                  (const coap_pkt_t *)&resp,
+                                  CONFIG_NANOCOAP_CACHE_RESPONSE_SIZE);
+
+    /* the absolute time of max-age should be at approx. now + 30 sec
+       (1 sec buffer) */
+    now = xtimer_now_usec();
```

Comment:
Here and below (and the unittest's Makefile)

```suggestion
    now = ztimer_now(ZTIMER_SEC);
```

no? There is no relation between `xtimer_now_usec()` (or `ztimer_now(ZTIMER_USEC)` for that matter) to `ztimer_now(ZTIMER_SEC)` and the latter is used internally to the cache... so comparing two timers who have no common time basis, is non-sensical in the tests (and might even lead to spurious failures).

Reply:
See also https://github.com/RIOT-OS/RIOT/pull/13889/files#r827160244